resolwe/flow/tests/test_relations.py

```diff
@@ -469,9 +513,15 @@ def test_delete_superuser(self):
         self.assertEqual(resp.status_code, status.HTTP_204_NO_CONTENT)
         self.assertFalse(Relation.objects.filter(pk=self.relation_group.pk).exists())
 
+    def test_delete_publicuser(self):
+        """Anonymous user must not be able to delete relations."""
+        resp = self._delete(self.relation_group.pk, user=AnonymousUser())
+        self.assertEqual(resp.status_code, status.HTTP_401_UNAUTHORIZED)
+        self.assertTrue(Relation.objects.filter(pk=self.relation_group.pk).exists())
+
     def test_delete_different_user(self):
         resp = self._delete(self.relation_group.pk, user=self.user)
-        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(resp.status_code, status.HTTP_401_UNAUTHORIZED)
```

Comment:
Does the user have the view permission on the object? If not, 404 should be returned.

Reply:
He does not have the permission. I will change the returned code.

But is it not more appropriate to return unauthorized in this case? It is true that more information is revealed (the existence of object, for example) but I believe it is nice to return unauthorized when someone does not have the permission to view the object.